packages/shell-api/src/helpers.ts

```diff
@@ -644,10 +668,83 @@ export async function getPrintableShardStatus(
     )
   ).filter((dbEntry) => !!dbEntry);
 
+  let shardedDataDistribution: ShardedDataDistribution | undefined;
+
+  try {
+    // Available since 6.0.3
+    const adminDb = db.getSiblingDB('admin');
+    shardedDataDistribution = (await (
+      await adminDb.aggregate([{ $shardedDataDistribution: {} }])
+    ).toArray()) as ShardedDataDistribution;
+  } catch (e) {
+    // Pass, most likely an older version.
+    console.error('shardedDataDistribution Errored:', e);
+  }
+
+  result.shardedDataDistribution = shardedDataDistribution;
+
   delete result.shardingVersion.currentVersion;
   return result;
 }
 
+type ShardingStatusResult = {
+  shardingVersion: {
+    _id: number;
+    clusterId: ObjectId;
+    /** This gets deleted when it is returned from getPrintableShardStatus */
+    currentVersion?: number;
+  };
+  shards: {
+    _id: string;
+    host: string;
+    state: number;
+    topologyTime: Timestamp;
+    replSetConfigVersion: Long;
+  }[];
+  [mongoses: `${string} mongoses`]:
+    | 'none'
+    | {
+        [version: string]:
+          | number
+          | {
+              up: number;
+              waiting: boolean;
+            };
+      }[];
+  autosplit: {
+    'Currently enabled': 'yes' | 'no';
+  };
+  balancer: {
+    'Currently enabled': 'yes' | 'no';
+    'Currently running': 'yes' | 'no' | 'unknown';
+    'Failed balancer rounds in last 5 attempts': number;
+    'Migration Results for the last 24 hours':
+      | 'No recent migrations'
+      | {
+          [count: number]:
+            | 'Success'
+            | `Failed with error '${string}', from ${string} to ${string}`;
+        };
+    'Balancer active window is set between'?: `${string} and ${string} server local time`;
+    'Last reported error'?: string;
+    'Time of Reported error'?: string;
+    'Collections with active migrations'?: `${string} started at ${string}`[];
+  };
+  shardedDataDistribution?: ShardedDataDistribution;
```

Comment:
I am adding it as a separate field at the moment since that's just a simple solution, but we could make it cleaner/nicer for the user. Open to suggestions